### PR TITLE
Upgrade zlib-rs to 0.3.0 to get multiple bugfixes

### PR DESCRIPTION
**Pull request by Shnatsel**
Fixes at least these correctness issues:

https://github.com/memorysafety/zlib-rs/pull/165
https://github.com/memorysafety/zlib-rs/issues/169
https://github.com/memorysafety/zlib-rs/issues/172

And possibly others.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -21,7 +21,7 @@ exclude = [".*"]
 [dependencies]
 libz-sys = { version = "1.1.20", optional = true, default-features = false }
 libz-ng-sys = { version = "1.1.16", optional = true }
-libz-rs-sys = { version = "0.2.1", optional = true, default-features = false, features = ["std", "rust-allocator"] }
```
Comment (Byron) [inline Cargo.toml:24]: Is there a good reason for removing the explicit handling of features? It looks like [they didn't change](https://docs.rs/crate/libz-rs-sys/latest/features).

The rationale for having them is to not rely on default feature remaining stable but instead take note consciously on which features are enabled.
I'd prefer to keep that.

**Review by Byron — CHANGES_REQUESTED** [review]
Thanks a lot!

Could you also add GitHub links (essentially the body of the PR) into the commit?

**Comment by Shnatsel** [comment]
I believe the merge commit would have PR description as the commit message anyway. But sure, done.

File: Cargo.toml [hunk 2]
```diff
@@ -21,7 +21,7 @@ exclude = [".*"]
...
+libz-rs-sys = { version = "0.3.0", optional = true, default-features = false, features = ["std", "rust-allocator"] } # this matches the default features, but we don't want to depend on the default features staying the same
```
Comment (Byron) [inline Cargo.toml:24]:
```suggestion
# this matches the default features, but we don't want to depend on the default features staying the same
libz-rs-sys = { version = "0.3.0", optional = true, default-features = false, features = ["std", "rust-allocator"] }
```

**Review by Byron — COMMENTED** [review]
Thanks a lot, this should do it.

If merge-commits would come up in code-lens or similar tools, that should be enough as well. I am not sure they do though.

Would you consider applying the suggestion? Once again I am too afraid to do it myself as it might prevent merging 😅.

**Comment by Shnatsel** [comment]
I assume @Byron will bump it as part of the usual release process. If not, I'm happy to increment that number, sure.

**Comment by Byron** [comment]
Actually, changing the version number will make a release more straightforward. It's a bit like being a genie, limitless power, but impossible to use for yourself :D.

**Review by Byron — APPROVED** [review]

**Comment by Shnatsel** [comment]
Version bumped

**Review by Byron — APPROVED** [review]
Thanks a lot!

**Pull request merged**